File: Easy/2025-11-04-3318-Find-X-Sum-of-All-K-Long-Subarrays-I/solution.js
var xSum = function(nums, k, x) {
    const n = nums.length;
    const answer = new Array(n - k + 1);
    
    for (let i = 0; i <= n - k; i++) {
        const subarray = nums.slice(i, i + k);
        const freqMap = new Map();
        
        for (const num of subarray) {
            freqMap.set(num, (freqMap.get(num) || 0) + 1);
        }
        
        const sortedFreq = Array.from(freqMap.entries())
            .sort((a, b) => b[1] - a[1] || b[0] - a[0])
            .slice(0, x);
        
        const sum = sortedFreq.reduce((acc, [num, count]) => acc + num * count, 0);
        answer[i] = sum;
    }
    
    return answer;
};

File: Easy/2025-11-04-3318-Find-X-Sum-of-All-K-Long-Subarrays-I/solution.py
from collections import Counter
from typing import List

def xSum(nums: List[int], k: int, x: int) -> List[int]:
    n = len(nums)
    answer = []
    
    for i in range(n - k + 1):
        subarray = nums[i:i + k]
        count = Counter(subarray)
        most_common = count.most_common(x)
        
        # If there are less than x distinct elements, sum all
        if len(most_common) < x:
            answer.append(sum(subarray))
        else:
            # Keep only the top x most frequent elements
            top_elements = sorted(most_common, key=lambda item: (-item[1], -item[0]))
            top_values = [val for val, freq in top_elements]
            result_sum = sum(val for val in subarray if val in top_values)
            answer.append(result_sum)
    
    return answer